### DISCUSSION TITLE
How to populate update form from table (external data source)

### DISCUSSION THREAD

**Discussion by helena-pa** [Budibase/budibase#9720] [Help] [answered]
Hello,

In a basic CRUD screen I want to populate an update form's fields from the selected row of a table.
I'm using Airtable as date source.

In my side panel with the update form I used my read query in the data provider and the update query in the form schema.
I set type to 'Update' in the form and auto generated the field group.
The fields are ok, my update query works, but the forms fields are not populated with the values's selected row. When I display the same values in a paragraph, it's work fine.

Is it possible to populate an update form like this? If so, how Budibase make the link between the read query providing data and the update query generating the form? 
If it's not possible like this, how to populate my update form? 
I'd also try a form set to 'Create' type and set form fields's default value with my repeater, and the displaying values are ok the first time, but when I click another row, the values stay the same (but the same value in a paragraph is well updated).

**Comment by melohagan** [comment, marked as answer]
Hey @helena-pa 

The [Form schema](https://docs.budibase.com/docs/forms#form-schema) cannot be automatically linked to anything that's not a SQL database or the internal Budibase DB.

When using Airtable (or any other custom query) you'll need to select a custom schema for the form and give the fields manually entered names.

There's an example [here](https://docs.budibase.com/docs/custom-queries#tutorial-posting-form-data-to-a-custom-query) of how to post a form to a custom query.

**Reply by helena-pa** [reply]
Hi @melohagan 

Ok thanks, I wasn't sure reading the documentation if an airtable query was a "custom query" (all the examples are about SQL custom queries) and I was able to select my airtable schema for the form, so I tried ^^

But I have the same problem when I select a custom schema for the form. You gave me an example to post a form to a custom query, but my custom query already works fine. My problem is for populating my update form.
The dcumentation say : "In Update mode, the form fields will automatically be populated from the parent Repeater data.", but my fields stay empty.

**Reply by melohagan** [reply]
Hey @helena-pa 

You're right that it's not clear that all the datasource queries are "custom queries". Maybe it's not the best name, I'll have a think about it.

So when it comes to custom queries, the *Update* mode of a form won't load in the values. The general problem is that a query by itself is not full CRUD functionality, so binding it to a form isn't necessarily going to be useful.

I had raised an issue here and there is a bit of discussion you can read through: https://github.com/Budibase/budibase/issues/6238

**Reply by helena-pa** [reply]
Ok I was beginning to suspect this answer ^^ Maybe this point could also be clarified in [the documentation](https://docs.budibase.com/docs/forms#form-type), because currently the warning on custom queries is only present in the "Form schema" section and not in the sister section "Form type" while it's possible to define the type of form with or without specific form schema :)

Ok so let's go back to my second solution: set the _Create_ mode of my form and set manually the _Default value_ for each form field with the repeater. Is this the right way to do it? 
Because when I try this, the displaying value in the fields in my side panel are ok the first time, but when I click another row, the values stay the same. However, the same binding in a paragraph works well.

First row click:
![image](https://user-images.githubusercontent.com/117634672/220063527-c22748b6-1873-4c35-8821-125694a12663.png)

Second row click:
![image](https://user-images.githubusercontent.com/117634672/220064533-f08e793c-5a9f-4d93-a858-75d8c55108e3.png)

**Reply by melohagan** [reply]
Hey @helena-pa 

It's because default values apply on load. When you have an open side panel and switch between the rows, the form fields aren't being re-loaded.

The workaround would be to use the **Update Field Value** action, which would be similar to this use case:
https://github.com/Budibase/budibase/discussions/9651#discussioncomment-4982255

**Reply by helena-pa** [reply]
It's works, and with this new structure I was able to refresh my table after the row update!
Thank you!

**Reply by melohagan** [reply]
You're welcome! 😃
